include/class_loader/class_loader_core.hpp

```diff
@@ -82,7 +80,12 @@ typedef std::string ClassName;
 typedef std::string BaseClassName;
 typedef std::map<ClassName, impl::AbstractMetaObjectBase *> FactoryMap;
 typedef std::map<BaseClassName, FactoryMap> BaseToFactoryMapMap;
-typedef std::pair<LibraryPath, Poco::SharedLibrary *> LibraryPair;
+
+#ifdef WIN32
+typedef std::pair<LibraryPath, HINSTANCE> LibraryPair;
+#else
+typedef std::pair<LibraryPath, void *> LibraryPair;
```

Comment:
Maybe use a typedef for the second type rather than having `LibraryPair` declared in two places (even if they're close to one another).